scripts/ci/metrics.sh

```diff
@@ -61,20 +51,49 @@ _create_job_record() {
     local commit_sha
     commit_sha="$(get_commit_sha)"
 
-    bq_create_job_record "$id" "$name" "$repo" "$branch" "$pr_number" "$commit_sha"
+    bq_create_job_record "$id" "$name" "$repo" "$branch" "$pr_number" "$commit_sha" "$ci_system"
+}
+
+_get_metrics_job_id() {
+    local id
+    if is_OPENSHIFT_CI; then
+        if [[ -z "${BUILD_ID:-}" ]]; then
+            info "Skipping job record for jobs without a BUILD_ID (bin, images)"
+            return
+        fi
+        id="${BUILD_ID}"
+    elif is_GITHUB_ACTIONS; then
+        # There's such thing as a unique GitHub Actions Job ID but it's not available to us.
+        # See https://github.com/orgs/community/discussions/8945
+        # We have to uniquely identify a job run differently. Here we use the following:
+        # * GITHUB_RUN_ID - workflow id, e.g. 14113014151.
+        # * GITHUB_RUN_ATTEMPT - workflow re-run attempt, e.g. 1, 2, 3, ... Because GITHUB_RUN_ID stays the same.
+        # * GITHUB_JOB - name of the job, e.g. "build-and-push-main".
+        # * A random number because the above is not enough to differentiate matrix jobs.
+        # The resulting value has to be stored for subsequent metrics updates.
+        id="${GITHUB_RUN_ID}.${GITHUB_RUN_ATTEMPT}.${GITHUB_JOB}.${RANDOM}"
```

Comment:
Can we "cache" the ID and then return it on every call of "_get_metrics_job_id"?

i.e.
```suggestion
    if [[ -z "${GHA_METRICS_JOB_ID:-}" ]]; then
        set_ci_shared_export "GHA_METRICS_JOB_ID" "${GITHUB_RUN_ID}.${GITHUB_RUN_ATTEMPT}.${GITHUB_JOB}.${RANDOM}"
    fi
    id="${GHA_METRICS_JOB_ID}"
```

In that case, we can just call `_get_metrics_job_id` where we need to, and we don't have to handle `METRICS_JOB_ID` in other places.

Reply:
I like it. Adopted.

Reply:
Good catch, thanks for taking a look. I'm trying to fix it now. Will ping for re-review.

Reply:
It looks better now.